src/vs/workbench/api/common/extHostChatAgents2.ts

```diff
@@ -36,10 +38,30 @@ export class ExtHostChatAgents2 implements ExtHostChatAgentsShape2 {
 
 	constructor(
 		mainContext: IMainContext,
+		commands: ExtHostCommands,
 		private readonly _extHostChatProvider: ExtHostChatProvider,
 		private readonly _logService: ILogService,
 	) {
 		this._proxy = mainContext.getProxy(MainContext.MainThreadChatAgents2);
+
+		commands.registerArgumentProcessor({
+			processArgument: (arg) => {
+				if (arg && arg.$mid === MarshalledId.ChatResponseContext) {
+					if (!arg.agentId || !arg.sessionId || !arg.requestId) {
+						return arg;
+					}
+
+					for (const [_, agent] of this._agents) {
+						if (agent.apiAgent.name === arg.agentId) {
+							const result = this._resultsBySessionAndRequestId.get(arg.sessionId)?.get(arg.requestId);
```

Comment:
I don't like this, these results aren't meant to be passed to random commands outside of the context of an agent, but as long as it's just proposed API for us